script.js

```diff
@@ -17,3 +17,210 @@ for (let i = 0; i < mobileItems.length; i += 1) {
+              </li>
+              <li>
+                  <a href="#">${element.technologies[2]}</a>
+              </li>
+          </ul>
+      </div>
+      <button type="submit" class="card-btn" id="card${element.id}">See Project</button>
+  </div>
+</div>`;
+}
+
+// popup window
+
+function getPopupId(id) {
+  for (let j = 0; j < projectsArr.length; j += 1) {
+    if (projectsArr[j].id === id) {
+      const popupWindow = document.getElementById('popup');
+      popupWindow.innerHTML += `<div class="popup-content">
+          <div class="popup-header">
+              <h2>${projectsArr[j].name}</h2>
+              <button class="cancle" id="cancle${projectsArr[j].id}"><img src="./images/Icon - Cancel.png" alt="cancle icon"></button>
+          </div>
+          <ul>
+              <li>
+                  <a href="#">${projectsArr[j].technologies[0]}</a>
+              </li>
+              <li>
+                  <a href="#">${projectsArr[j].technologies[1]}</a>
+              </li>
+              <li>
+                  <a href="#">${projectsArr[j].technologies[2]}</a>
+              </li>
+          </ul>
+          <div class="popup-footer">
+              <img class="popup-img" src="${projectsArr[j].featuredImg}" alt="">
+              <div class="footer-content">
+                  <p>${projectsArr[j].desc}</p>
+                  <div class="popup-btn">
+                      <a href="${projectsArr[j].liveLink}">See live <img src="./images/live.png" alt=""></a>
+                      <a href="${projectsArr[j].sourceLink}">See source <img src="./images/Icon -GitHub.png" alt=""></a>
+                  </div>
+              </div>
+          
+          </div>
+          </div>`;
+    }
+  }
+  const canclex = document.querySelector(`#cancle${id}`);
+  canclex.addEventListener('click', () => {
+    const popup = document.querySelector('.popup');
+    popup.style.display = 'none';
+  });
+}
+
+const primarybtn = document.querySelector('.primary-btn');
+primarybtn.addEventListener('click', () => {
+  const popup = document.querySelector('.popup');
+  popup.style.display = 'block';
+  popup.innerHTML = '';
+  getPopupId(workproject.id);
+});
+
+for (let i = 0; i < projectsArr.length; i += 1) {
+  const cardBtn = document.querySelector(`#card${projectsArr[i].id}`);
+  cardBtn.addEventListener('click', () => {
+    const popup = document.querySelector('.popup');
+    popup.style.display = 'block';
+    popup.innerHTML = '';
+    getPopupId(projectsArr[i].id);
+  });
+}
+
+// form validation
+
+const form = document.getElementById('contact-form');
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const email = document.getElementById('email').value;
+
+  const regex = /^((?!\.)[\w\-_.]*[^.])(@\w+)(\.\w+(\.\w+)?[^.\W])$/;
+
+  const formErrorMsg = document.querySelector('.form-error-msg');
+  if (!regex.test(email)) {
+    formErrorMsg.innerHTML = 'error';
+  } else {
+    formErrorMsg.innerHTML = 'success';
+  }
+});
```

Comment:
As requested in the project requirements:
![image](https://user-images.githubusercontent.com/42799579/209006766-f0f4cff5-ca03-4505-a89c-856d45e7fb6b.png)

You should display an error message if the email isn't in lowercase 😄! Currently, the success message is being displayed for both cases:

With upper case:
![image](https://user-images.githubusercontent.com/42799579/209006883-d02b2233-d811-498d-87f9-6acf54d70ea2.png)

With lowercase:
![image](https://user-images.githubusercontent.com/42799579/209006974-ed3ab638-8682-418a-8c43-bedaba98ed2b.png)

And the form isn't being submitted 😉. Please make sure to submit the form correctly if the email is in lowercase and display the corresponding error message if it contains an uppercase letter.